aws/resource_aws_lb_target_group_test.go

```diff
@@ -14,6 +16,69 @@ import (
 	"github.com/hashicorp/terraform/terraform"
 )
 
+func init() {
+	resource.AddTestSweepers("aws_lb_target_group", &resource.Sweeper{
+		Name: "aws_lb_target_group",
+		F:    testSweepLBTargetGroups,
+		Dependencies: []string{
+			"aws_lb",
+		},
+	})
+}
+
+func testSweepLBTargetGroups(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).elbv2conn
+
+	prefixes := []string{
+		"tf-",
+		"tf-test-",
+		"tf-acc-test-",
+		"test",
+	}
+
+	err = conn.DescribeTargetGroupsPages(&elbv2.DescribeTargetGroupsInput{}, func(page *elbv2.DescribeTargetGroupsOutput, isLast bool) bool {
+		if page == nil || len(page.TargetGroups) == 0 {
+			log.Print("[DEBUG] No LB Target Groups to sweep")
+			return false
+		}
+
+		for _, targetGroup := range page.TargetGroups {
+			name := aws.StringValue(targetGroup.TargetGroupName)
+			skip := true
+			for _, prefix := range prefixes {
+				if strings.HasPrefix(name, prefix) {
+					skip = false
+					break
+				}
+			}
+			if skip {
+				log.Printf("[INFO] Skipping LB Target Group: %s", name)
+				continue
+			}
```

Comment:
_minor optimization:_ you could move this inside by negating the `strings.HasPrefix(name, prefix)` method? e.g.

```
if !strings.HasPrefix(name, prefix) {
  log.Printf("[INFO] Skipping LB Target Group: %s", name)
  continue
}
```

Reply:
Since we're looping through multiple incorrect prefixes (only one will potentially be correct), we cannot short circuit looping through all the variations first 😄 

At some point we will fix all the test resource naming to be consistent though and the `for` loop will be moot! e.g. `acctest.RandomWithPrefix("tf-acc-test")`